dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/dataelement/DataElementServiceTest.java

```diff
@@ -98,6 +142,44 @@ void testUpdateDataElement()
         assertEquals( ValueType.BOOLEAN, dataElementA.getValueType() );
     }
 
+    @Test
+    void testUpdateDataElement_WithOptionSet()
+    {
+        DataElement de = createDataElementWithOptionSet( ValueType.MULTI_TEXT, ValueType.MULTI_TEXT );
+        dataElementService.addDataElement( de );
+        de.setName( "new-name" );
+        assertDoesNotThrow( () -> dataElementService.updateDataElement( de ) );
+    }
+
+    @Test
+    void testUpdateDataElement_WithOptionSetValueTypeMismatch()
+    {
+        DataElement de = createDataElementWithOptionSet( ValueType.MULTI_TEXT, ValueType.MULTI_TEXT );
+        dataElementService.addDataElement( de );
+        de.setValueType( ValueType.TEXT );
+
+        IllegalQueryException ex = assertThrows( IllegalQueryException.class,
+            () -> dataElementService.updateDataElement( de ) );
+        assertEquals( ErrorCode.E1115, ex.getErrorCode() );
+        assertEquals( "Data element value type must match option set value type: `MULTI_TEXT`", ex.getMessage() );
+    }
+
+    @Test
+    void testUpdateDataElement_WithOptionSetHavingCodeWithSeparator()
+    {
+        DataElement de = createDataElement( 'A' );
+        dataElementService.addDataElement( de );
+        de.setValueType( ValueType.MULTI_TEXT );
+        de.setOptionSet( addOptionSet( ValueType.MULTI_TEXT, "A", "B", "C,D" ) );
+
+        IllegalQueryException ex = assertThrows( IllegalQueryException.class,
+            () -> dataElementService.updateDataElement( de ) );
+        assertEquals( ErrorCode.E1117, ex.getErrorCode() );
```

Comment:
Very minor, we have `org.hisp.dhis.utils.Assertions.assertThrowsErrorCode` which could make these error code assertions a bit more compact.

Reply:
nice, will use it where I don't check the message as well